The job **`x86_64-gnu-llvm-9`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/2033166619) [(plain)](https://github.com/rust-lang/rust/commit/0a8e401188062f0c60c989978352663b1e25e70e/checks/2033166619/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.................................................................................................... 9300/11526
.................................................................................................... 9400/11526
...........................................................................i......i................. 9500/11526
.................................................................................................... 9600/11526
..............iiiiiii..iiiiii.i..................................................................... 9700/11526
.................................................................................................... 9900/11526
.................................................................................................... 10000/11526
.................................................................................................... 10100/11526
.................................................................................................... 10200/11526
---
Suite("src/test/assembly") not skipped for "bootstrap::test::Assembly" -- not in ["src/tools/tidy"]
Check compiletest suite=assembly mode=assembly (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)

running 29 tests
iiiiiiiiiiiiiiiiiiiiiiiiiiiii

Suite("src/test/incremental") not skipped for "bootstrap::test::Incremental" -- not in ["src/tools/tidy"]
 finished in 0.067 seconds
Check compiletest suite=incremental mode=incremental (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
---
Suite("src/test/debuginfo") not skipped for "bootstrap::test::Debuginfo" -- not in ["src/tools/tidy"]
Check compiletest suite=debuginfo mode=debuginfo (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)

running 116 tests
iiiiiiiiii.i.i...i.i..ii....i.i....ii..........iiii.........i.....i...i.......ii.i.ii.....iiii.....i 100/116
test result: ok. 78 passed; 0 failed; 38 ignored; 0 measured; 0 filtered out; finished in 2.56s

 finished in 2.637 seconds
Suite("src/test/ui-fulldeps") not skipped for "bootstrap::test::UiFullDeps" -- not in ["src/tools/tidy"]
---
.................................................................................................... 200/2842
..ii................................................................................................ 300/2842
...................................................i................................................ 400/2842
.................................................................................................... 500/2842
........................i..i...........F.....F.iiii................................................. 600/2842
.................................................................................................... 800/2842
.................................................................................................... 900/2842
.................................................................................................... 1000/2842
.................................................................................................... 1100/2842
---
---- src/macros/mod.rs - assert_matches (line 126) stdout ----
error[E0658]: use of unstable library feature 'assert_matches'
 --> src/macros/mod.rs:129:1
  |
6 | assert_matches!(a, Some(_));
  |
  |
  = help: add `#![feature(assert_matches)]` to the crate attributes to enable
error[E0658]: use of unstable library feature 'assert_matches'
 --> src/macros/mod.rs:130:1
  |
  |
7 | assert_matches!(b, None);
  |
  |
  = help: add `#![feature(assert_matches)]` to the crate attributes to enable
error[E0658]: use of unstable library feature 'assert_matches'
  --> src/macros/mod.rs:133:1
   |
   |
10 | assert_matches!(a, Ok(x) | Err(x) if x.len() < 100);
   |
   |
   = help: add `#![feature(assert_matches)]` to the crate attributes to enable
error[E0308]: mismatched types
  --> src/macros/mod.rs:133:20
   |
   |
10 | assert_matches!(a, Ok(x) | Err(x) if x.len() < 100);
   |                 -  ^^^^^ expected enum `Option`, found enum `Result`
   |                 this expression has type `Option<u32>`
   |
   = note: expected enum `Option<u32>`
              found enum `Result<_, _>`
              found enum `Result<_, _>`

error[E0308]: mismatched types
  --> src/macros/mod.rs:133:28
   |
10 | assert_matches!(a, Ok(x) | Err(x) if x.len() < 100);
   |                 -          ^^^^^^ expected enum `Option`, found enum `Result`
   |                 this expression has type `Option<u32>`
   |
   = note: expected enum `Option<u32>`
              found enum `Result<_, _>`
---
---- src/macros/mod.rs - debug_assert_matches (line 281) stdout ----
error[E0658]: use of unstable library feature 'assert_matches'
 --> src/macros/mod.rs:284:1
  |
6 | debug_assert_matches!(a, Some(_));
  |
  |
  = help: add `#![feature(assert_matches)]` to the crate attributes to enable
error[E0658]: use of unstable library feature 'assert_matches'
 --> src/macros/mod.rs:285:1
  |
  |
7 | debug_assert_matches!(b, None);
  |
  |
  = help: add `#![feature(assert_matches)]` to the crate attributes to enable
error[E0658]: use of unstable library feature 'assert_matches'
  --> src/macros/mod.rs:288:1
   |
   |
10 | debug_assert_matches!(a, Ok(x) | Err(x) if x.len() < 100);
   |
   |
   = help: add `#![feature(assert_matches)]` to the crate attributes to enable
error[E0308]: mismatched types
  --> src/macros/mod.rs:288:26
   |
   |
10 | debug_assert_matches!(a, Ok(x) | Err(x) if x.len() < 100);
   |                       -  ^^^^^ expected enum `Option`, found enum `Result`
   |                       this expression has type `Option<u32>`
   |
   = note: expected enum `Option<u32>`
              found enum `Result<_, _>`
              found enum `Result<_, _>`

error[E0308]: mismatched types
  --> src/macros/mod.rs:288:34
   |
10 | debug_assert_matches!(a, Ok(x) | Err(x) if x.len() < 100);
   |                       -          ^^^^^^ expected enum `Option`, found enum `Result`
   |                       this expression has type `Option<u32>`
   |
   = note: expected enum `Option<u32>`
              found enum `Result<_, _>`
---

error: test failed, to rerun pass '--doc'


command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "--target" "x86_64-unknown-linux-gnu" "-Zbinary-dep-depinfo" "-j" "16" "--release" "--locked" "--color" "always" "--features" "panic-unwind backtrace compiler-builtins-c" "--manifest-path" "/checkout/library/test/Cargo.toml" "-p" "core" "--" "--quiet"


failed to run: /checkout/obj/build/bootstrap/debug/bootstrap --stage 2 test --exclude src/tools/tidy
Build completed unsuccessfully in 0:20:57
```

</details>

--- a/library/core/src/macros/mod.rs
+++ b/library/core/src/macros/mod.rs
@@ -141,9 +141,9 @@ macro_rules! assert_matches {
     ($left:expr, $( $pattern:pat )|+ $( if $guard: expr )? $(,)?) => ({
         match $left {
             $( $pattern )|+ $( if $guard )? => {}
-            left_val => {
+            ref left_val => {
                 $crate::panicking::assert_matches_failed(
-                    &left_val,
+                    left_val,
                     $crate::stringify!($($pattern)|+ $(if $guard)?),
                     $crate::option::Option::None
                 );
@@ -153,9 +153,9 @@ macro_rules! assert_matches {
     ($left:expr, $( $pattern:pat )|+ $( if $guard: expr )?, $($arg:tt)+) => ({
         match $left {
             $( $pattern )|+ $( if $guard )? => {}
-            left_val => {
+            ref left_val => {
                 $crate::panicking::assert_matches_failed(
-                    &left_val,
+                    left_val,
                     $crate::stringify!($($pattern)|+ $(if $guard)?),
                     $crate::option::Option::Some($crate::format_args!($($arg)+))
                 );

--- a/library/core/src/macros/mod.rs
+++ b/library/core/src/macros/mod.rs
@@ -124,13 +124,15 @@ macro_rules! assert_ne {
 /// # Examples
 ///
 /// ```
+/// #![feature(assert_matches)]
+///
 /// let a = 1u32.checked_add(2);
 /// let b = 1u32.checked_sub(2);
 /// assert_matches!(a, Some(_));
 /// assert_matches!(b, None);
 ///
 /// let c = Ok("abc".to_string());
-/// assert_matches!(a, Ok(x) | Err(x) if x.len() < 100);
+/// assert_matches!(c, Ok(x) | Err(x) if x.len() < 100);
 /// ```
 #[macro_export]
 #[unstable(feature = "assert_matches", issue = "none")]
@@ -279,13 +281,15 @@ macro_rules! debug_assert_ne {
 /// # Examples
 ///
 /// ```
+/// #![feature(assert_matches)]
+///
 /// let a = 1u32.checked_add(2);
 /// let b = 1u32.checked_sub(2);
 /// debug_assert_matches!(a, Some(_));
 /// debug_assert_matches!(b, None);
 ///
 /// let c = Ok("abc".to_string());
-/// debug_assert_matches!(a, Ok(x) | Err(x) if x.len() < 100);
+/// debug_assert_matches!(c, Ok(x) | Err(x) if x.len() < 100);
 /// ```
 #[macro_export]
 #[unstable(feature = "assert_matches", issue = "none")]

--- a/library/core/src/macros/mod.rs
+++ b/library/core/src/macros/mod.rs
@@ -135,7 +135,7 @@ macro_rules! assert_ne {
 /// assert_matches!(c, Ok(x) | Err(x) if x.len() < 100);
 /// ```
 #[macro_export]
-#[unstable(feature = "assert_matches", issue = "none")]
+#[unstable(feature = "assert_matches", issue = "82775")]
 #[allow_internal_unstable(core_panic)]
 macro_rules! assert_matches {
     ($left:expr, $( $pattern:pat )|+ $( if $guard: expr )? $(,)?) => ({
@@ -292,7 +292,7 @@ macro_rules! debug_assert_ne {
 /// debug_assert_matches!(c, Ok(x) | Err(x) if x.len() < 100);
 /// ```
 #[macro_export]
-#[unstable(feature = "assert_matches", issue = "none")]
+#[unstable(feature = "assert_matches", issue = "82775")]
 #[allow_internal_unstable(assert_matches)]
 macro_rules! debug_assert_matches {
     ($($arg:tt)*) => (if $crate::cfg!(debug_assertions) { $crate::assert_matches!($($arg)*); })

